src/components/FlowbiteIllustration.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+export const FlowbiteIllustration = ({
+	name,
+	className = '', 
+	width = '300px',
+	height = 'auto',
+	title = '',
+	basePath = '/media/illustrations',
+}) => {
+	// Sanitize the basePath to prevent path traversal
+	const sanitizedBasePath = basePath
+		?.replace(/\.\./g, '') // Remove path traversal attempts
+		?.replace(/\/+/g, '/') // Replace multiple slashes with single slash
+		?.replace(/\/$/, ''); // Remove trailing slash
+
+	// Sanitize the name to prevent path traversal and ensure valid filename
+	const sanitizedName = name
+		?.replace(/[^a-zA-Z0-9\-_]/g, '') // Only allow alphanumeric, hyphens, underscores
+		?.toLowerCase(); // Normalize to lowercase
+
+	if (!sanitizedBasePath) {
+		console.warn(`Invalid base path: '${basePath}'`);
+		return null;
+	}
+
+	if (!sanitizedName) {
+		console.warn(`Invalid or empty illustration name: '${name}'`);
+		return null;
+	}	
+
+	const illustrationSrc = `${basePath}/${sanitizedName}.svg`;
+
+	return (
+		<div className='text-center'>
+			<img 
+				src={illustrationSrc}
+				alt={title || `${name} illustration`}
+				title={title}
+				className={className}
+				style={{ 
+					width, 
+					height,
+					maxWidth: '100%',
```

Comment:
I have some concerns here, if the `width` will be passed as a prop in `%` ie `width='80%'`, then setting maxWidth to 100% may cause some unexpected behavior.

Honestly, I would recommend to use `width` and `height` to be by default set to `100%` (and remove the `maxWidth`, then the container (where the FlowBiteIllustration` component is used should take care of it and prevent the overflowing etc.